tools/options.c

```diff
@@ -366,7 +369,8 @@ static void nn_append_string_to_free (struct nn_parse_context *ctx,
     if (!lst->items) {
         nn_memory_error (ctx);
     }
-    lst->to_free[lst->to_free_num-1] = str;
+    nn_assert (lst->to_free);
```

Comment:
I'm withdrawing all my exceptions to this usage in this file.  This is in options.c -- so not in the mainline library, and the assertions therefore do no harm to any end users.  The need for them underscores pretty strongly to me that the options design here is somewhat sketchy, but it works for now.